agent/src/com/jonnyzzz/teamcity/plugins/node/agent/NodeDetector.kt

```diff
@@ -108,4 +130,53 @@ class NodeToolsDetector(events: EventDispatcher<AgentLifeCycleListener>,
       }
     })
   }
+
+  private fun setVerbose(verbose: Boolean) {
+    // TODO: either modify logger or add some boolean field which used before all LOG calls
+  }
+
+  private fun locateInstalledNVM(): String? {
+    if (!(config.systemInfo.isUnix || config.systemInfo.isMac)) {
+      LOG.info("NVM supported only on Unix and Mac OSX")
+      return null
+    }
+
+    val nvm_home = System.getenv("NVM_HOME")
+    if (nvm_home != null && isNVMInstallation(File(nvm_home))) return nvm_home
+
+    val home = System.getProperty("user.home")
+    if (home != null) {
+      val candidate = File(home, ".nvm")
+      if (isNVMInstallation(candidate)) return candidate.absolutePath
+    }
+
+    return null
```

Comment:
Why not to have a configuration parameter in `buildAgent.properties` to override default search path?